tests/unit/misc/test_cmdhistory.py

```diff
@@ -0,0 +1,196 @@
+        for i in self.hist._tmphist:
+            assert i in HISTORY
+
+        for i in HISTORY:
+            assert i in self.hist._tmphist
+
+    def test_start_no_items(self):
+        """Test start with no matching text."""
+        with pytest.raises(HistoryEmptyError) as excinfo:
+            self.hist.start('k')
+        assert str(excinfo.value) == "History is empty."
+        assert not self.hist._tmphist
+
+    def test_get_item(self):
+        """Test __get_item__."""
+        for i in range(0, len(HISTORY)):
+            assert self.hist[i] == HISTORY[i]
+
+    def test_not_browsing_error(self):
+        """Test that next/previtem throws a ValueError"""
+        with pytest.raises(ValueError) as error1:
+            self.hist.nextitem()
+        assert str(error1.value) == "Currently not browsing history"
+
+        with pytest.raises(ValueError) as error2:
+            self.hist.previtem()
+        assert str(error2.value) == "Currently not browsing history"
+
+    def return_item(self):
+        return 'item'
+
+    def test_nextitem_single(self):
+        """Test nextitem() with valid input."""
+        self.hist.start('f')
+        self.hist._tmphist.nextitem = self.return_item
+        assert self.hist.nextitem() == 'item'
+
+    def test_previtem_single(self):
+        """Test previtem() with valid input."""
+        self.hist.start('f')
+        self.hist._tmphist.previtem = self.return_item
+        assert self.hist.previtem() == 'item'
+
+    def test_nextitem_previtem_chain(self):
+        """Test a combination of nextitem and previtem statements"""
+        assert self.hist.start('f') == 'fifth'
+        assert self.hist.previtem() == 'fourth'
+        assert self.hist.previtem() == 'first'
+        assert self.hist.nextitem() == 'fourth'
+
+    def raise_index_error(self):
+        raise IndexError()
+
+    def test_nextitem_index_error(self):
+        """"Test nextitem() when _tmphist raises an IndexError"""
+        self.hist.start('f')
+        self.hist._tmphist.nextitem = self.raise_index_error
+        with pytest.raises(HistoryEndReachedError) as excinfo:
+            self.hist.nextitem()
+        assert str(excinfo.value) == "History end reached"
+
+    def test_previtem_index_error(self):
+        """"Test previtem() when _tmphist raises an IndexError"""
+        self.hist.start('f')
+        self.hist._tmphist.previtem = self.raise_index_error
+        with pytest.raises(HistoryEndReachedError) as excinfo:
+            self.hist.previtem()
+        assert str(excinfo.value) == "History end reached"
+
+    def test_append_private_mode(self, monkeypatch, config_stub):
+        """Test append in private mode."""
+        self.hist.handle_private_mode = True
+        # We want general.private-browsing set to True
+        config_stub.data = CONFIG_PRIVATE
+        monkeypatch.setattr('qutebrowser.misc.cmdhistory.config',
+                            config_stub)
```

Comment:
Like with the `basekeyparser` tests, I think the monkeypatching in those functions isn't necessary as `config_stub` already takes care of that.